src/test/unit_tests/test_nsfs_access.js

```diff
@@ -18,8 +18,11 @@ mocha.describe('new tests check', async function() {
     const p = '/tmp/dir/';
     const root_dir = 'root_dir';
     const non_root_dir = 'non_root_dir';
+    const non_root_dir2 = 'non_root_dir2';
     const full_path_root = path.join(p, root_dir);
-    const full_path_non_root = path.join(p, non_root_dir);
+    const full_path_non_root = path.join(full_path_root, non_root_dir);
```

Comment:
Why did you change it? Now it is `/tmp/dir/root_dir/non_root_dir`?
```suggestion
    const full_path_non_root = path.join(p, non_root_dir);
```

Reply:
because in order that my new tests will work i needed the root directory to have full permissions. but the existing test expect it to have partial permissions. so I moved it into a folder with partial permissions. maybe I could do it reverse where my tests were part of different directory, but I don't think it matters that much